rmf_site_editor/src/interaction/select/select_anchor.rs

```diff
@@ -0,0 +1,650 @@
+        minimum_points: usize,
+        allow_inner_loops: bool,
+        implied_complete_loop: bool,
+        scope: AnchorScope,
+    ) {
+        let state = self
+            .commands
+            .spawn(SelectorInput(CreatePath::new(
+                spawn_path,
+                minimum_points,
+                allow_inner_loops,
+                implied_complete_loop,
+                scope,
+            )))
+            .id();
+
+        self.send(RunSelector {
+            selector: self.services.create_path,
+            input: Some(state),
+        });
+    }
+
+    pub fn create_point<T: Bundle + From<Point<Entity>>>(
+        &mut self,
+        repeating: bool,
+        scope: AnchorScope,
+    ) {
+        let state = self
+            .commands
+            .spawn(SelectorInput(CreatePoint::new::<T>(repeating, scope)))
+            .id();
+
+        self.send(RunSelector {
+            selector: self.services.create_point,
+            input: Some(state),
+        });
+    }
+
+    pub fn replace_point(&mut self, point: Entity, scope: AnchorScope) {
+        let state = self
+            .commands
+            .spawn(SelectorInput(ReplacePoint::new(point, scope)))
+            .id();
+
+        self.send(RunSelector {
+            selector: self.services.replace_point,
+            input: Some(state),
+        });
+    }
+
+    fn send(&mut self, run: RunSelector) {
+        self.commands.add(move |world: &mut World| {
+            world.send_event(run);
+        });
+    }
+}
+
+#[derive(Resource, Default)]
+pub struct HiddenSelectAnchorEntities {
+    /// All drawing anchors, hidden when users draw level entities such as walls, lanes, floors to
+    /// make sure they don't connect to drawing anchors
+    pub drawing_anchors: HashSet<Entity>,
+}
+
+/// The first five services should be customized for the State data. The services
+/// that return NodeResult should return `Ok(())` if it is okay for the
+/// workflow to continue as normal, and they should return `Err(None)` if it's
+/// time for the workflow to terminate as normal. If the workflow needs to
+/// terminate because of an error, return `Err(Some(_))`.
+pub fn build_anchor_selection_workflow<State: 'static + Send + Sync>(
+    anchor_setup: Service<BufferKey<State>, SelectionNodeResult>,
+    state_setup: Service<BufferKey<State>, SelectionNodeResult>,
+    update_preview: Service<(Hover, BufferKey<State>), SelectionNodeResult>,
+    update_current: Service<(SelectionCandidate, BufferKey<State>), SelectionNodeResult>,
+    handle_key_code: Service<(KeyCode, BufferKey<State>), SelectionNodeResult>,
+    cleanup_state: Service<BufferKey<State>, SelectionNodeResult>,
+    anchor_cursor_transform: Service<(), ()>,
+    anchor_select_stream: Service<(), (), (Hover, Select)>,
+    keyboard_just_pressed: Service<(), (), StreamOf<KeyCode>>,
+    cleanup_anchor_selection: Service<(), ()>,
+) -> impl FnOnce(Scope<Option<Entity>, ()>, &mut Builder) {
```

Comment:
This is also a fairly complex function, maybe a breakdown of what each service that users are expected to set is supposed to do would help?